# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/chrome-webidl/FrameLoader.webidl
# Commit: f95c19e45ab2
# Full Hash: f95c19e45ab2ef6a1ba8ac8d4124df117928b1a4
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-22 21:55:45
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r c749c42dc6b9 -r f95c19e45ab2 dom/chrome-webidl/FrameLoader.webidl
--- a/dom/chrome-webidl/FrameLoader.webidl	Wed Jul 22 18:40:32 2020 +0300
+++ b/dom/chrome-webidl/FrameLoader.webidl	Wed Jul 22 15:54:01 2020 +0000
@@ -122,12 +122,11 @@
    * @param aOuterWindowID the ID of the outer window to print
    * @param aPrintSettings optional print settings to use; printSilent can be
    *                       set to prevent prompting.
-   * @param aProgressListener optional print progress listener.
+   * @return A Promise that resolves once printing is finished.
    */
   [Throws]
-  void print(unsigned long long aOuterWindowID,
-             nsIPrintSettings aPrintSettings,
-             optional nsIWebProgressListener? aProgressListener = null);
+  Promise<void> print(unsigned long long aOuterWindowID,
+                      nsIPrintSettings aPrintSettings);
 
   /**
    * The element which owns this frame loader.